File: algorithms/0121.best_time_to_buy_and_sell_stock/best_time_to_buy_and_sell_stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit = 0;
        if prices.is_empty() {
            return profit;
        }
        let mut min_price = prices[0];
        for &price in prices.iter() {
            min_price = min_price.min(price);
            profit = profit.max(price - min_price);
        }
        profit
    }
}

File: algorithms/0121.best_time_to_buy_and_sell_stock/best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int min_price = prices[0];
        for (const int& price: prices) {
            min_price = min(price, min_price);
            profit = max(price - min_price, profit);
        }
        return profit;
    }
};